### DISCUSSION TITLE
minimal postgres version?

### DISCUSSION THREAD

**Discussion by glushakov** [weaponry/pgscv#22] [General]
Hi.
What is the minimal postgres version needed to run exporter?
I tryied to start it working with pg 9.5

#root /usr/bin/pgscv --config-file=/etc/pgscv9.yaml --log-level="debug"
{"level":"info","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"read configuration from /etc/pgscv9.yaml"}
{"level":"info","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"no-track disabled, for details check the documentation about 'no_track_mode' option."}
{"level":"debug","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"compile filters"}
{"level":"debug","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"filters compiled successfully"}
{"level":"debug","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"start application"}
{"level":"debug","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"config: add services from configuration"}
{"level":"info","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"registered new service [system:0]"}
{"level":"info","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"registered new service [postgres95]"}
{"level":"debug","service":"pgscv","time":"2021-11-19T12:36:43+03:00","message":"service [postgres95] available through: pgscv_user@postgres95:5432/postgres"}

Last string in log about that service successfully registered, but when i tryed to get metrics via CURL  - there are only system metrics.
I tried also disable all postgres collectors , except custom SQL query - and nothing changes

**Comment by lesovsky** [comment]
I tested pgscv with postgres 9.x and newer. But there are some important points:
- older Postgres provide lesser amount of metrics
- I follow Postgres versioning policy and don't guarantee proper work on versions which are not supported.
- some collectors might not work because there is no required functions in Postgres (see Availability column in [Collectors](https://github.com/weaponry/pgscv/wiki/Collectors) table.

Ad-hoc example with 9.0
```
lesovsky@matanuii:~  $ curl -s http://127.0.0.1:19890/metrics |grep -c ^postgres
207
lesovsky@matanuii:~  $ curl -s http://127.0.0.1:19890/metrics |grep version
go_info{version="go1.17"} 1
postgres_service_settings_info{name="server_version",service_id="postgres",setting="9.0.22",source="main",unit="",vartype="string"} 0
postgres_service_settings_info{name="server_version_num",service_id="postgres",setting="90022",source="main",unit="",vartype="integer"} 90022
```

**Comment by glushakov** [comment]
It looks like problem with permission.
When i  granted superuser to pgscv_user - metrics becomes available.
I logged all query from pgscv_user and find that exporter read settings: block_size, wal_block_size,wal_segment_size, server_version_num and data_directory.
And setting  data_directory available only for superuser (or member of pg_read_all_settings which not exist on 9.5 )
Maybe problem in this ?

**Reply by lesovsky** [reply]
yes, sure, the problem is insufficient privileges.

**Reply by glushakov** [reply]
ok, thanks
